src/mail/view/MailViewer.js

```diff
@@ -835,10 +852,16 @@ export class MailViewer {
 
 	/** @return list of inline referenced cid */
 	_loadMailBody(mail: Mail): Promise<Array<string>> {
-		return this._entityClient.load(MailBodyTypeRef, mail.body).then(body => {
+		return Promise.all([
+			this._entityClient.load(MailBodyTypeRef, mail.body),
+			worker.isAllowedExternalSender(mail.sender.address)
+		]).spread((body, isAllowedExternalSender) => {
```

Comment:
No `spread()`s. Its unsafe. I remove it in the other PR. Use proper argument destructuring.